components/server/ee/src/billing/entitlement-service.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { UserDB } from "@gitpod/gitpod-db/lib";
+import {
+    User,
+    WorkspaceInstance,
+    WorkspaceTimeoutDuration,
+    WORKSPACE_TIMEOUT_DEFAULT_LONG,
+    WORKSPACE_TIMEOUT_DEFAULT_SHORT,
+} from "@gitpod/gitpod-protocol";
+import { LicenseEvaluator } from "@gitpod/licensor/lib";
+import { Feature } from "@gitpod/licensor/lib/api";
+import { inject, injectable } from "inversify";
+import { BillingModes } from "../../../src/billing/billing-mode";
+import { EntitlementService } from "../../../src/billing/entitlement-service";
+import { MayStartWorkspaceResult } from "../user/eligibility-service";
+import { EntitlementServiceUBB } from "./entintlement-service-ubb";
+import { EntitlementServiceChargebee } from "./entitlement-service-chargebee";
+
+@injectable()
+export class EntitlementServiceImpl implements EntitlementService {
+    @inject(BillingModes) protected readonly billingModes: BillingModes;
+    @inject(EntitlementServiceChargebee) protected readonly chargebee: EntitlementServiceChargebee;
+    @inject(EntitlementServiceUBB) protected readonly ubb: EntitlementServiceUBB;
+    @inject(LicenseEvaluator) protected readonly licenseEvaluator: LicenseEvaluator;
+    @inject(UserDB) protected readonly userDb: UserDB;
+
+    async mayStartWorkspace(
+        user: User,
+        date: Date,
+        runningInstances: Promise<WorkspaceInstance[]>,
+    ): Promise<MayStartWorkspaceResult> {
+        const billingMode = await this.billingModes.getBillingModeForUser(user);
+        switch (billingMode.mode) {
+            case "none":
```

Comment:
For my own (and future reviewers) sanity, The mode of `none` only gets set for `Self-hosted`, bacause it depends on the `paymentEnabled` config to be false.